### ISSUE TITLE
Kustomize: unable to patch multiple resources due to field requirement

### ISSUE THREAD

**Issue by brenix** [holos-run/holos#348]
Kustomize has the ability to [patch multiple resources](https://github.com/kubernetes-sigs/kustomize/blob/master/examples/patchMultipleObjects.md)

As an example, you can target patch all CustomResourceDefinitions by specifying just the target as so:
```yaml
patches:
- target:
    kind: CustomResourceDefinition
  patch: ...
```

In CUE, this is what I have as an example:
```cue
KustomizeConfig: Kustomization: patches: [
  {
    target: kind: "CustomResourceDefinition"
    patch: yaml.Marshal([{
      op: "add"
      path: "/metadata/annotations/example"
      value: "example-value"
    }])
  }
]
```

An attempt to render a component with the patch above via holos results in a failure;
```
could not run: cue: marshal error: holos.spec.artifacts.0.transformers.0.kustomize.kustomization.patches.0.target.name: cannot convert incomplete value "string" to JSON at holos/holos/holos.go:106
```

This appears to be due to the generated kustomize types. If I manually edit the following file and set the name field as optional, I am able to render the component properly

```diff
diff --git a/cue.mod/gen/sigs.k8s.io/kustomize/api/types/var_go_gen.cue b/cue.mod/gen/sigs.k8s.io/kustomize/api/types/var_go_gen.cue
index a09ac50..c378a02 100644
--- a/cue.mod/gen/sigs.k8s.io/kustomize/api/types/var_go_gen.cue
+++ b/cue.mod/gen/sigs.k8s.io/kustomize/api/types/var_go_gen.cue
@@ -30,7 +30,7 @@ package types
 // for Var.ObjRef
 #Target: {
        apiVersion?: string @go(APIVersion)
-       name:        string @go(Name)
+       name?:       string @go(Name)
        namespace?:  string @go(Namespace)
 }
```

I'm not sure on what the solution looks like here, but wanted to report it as it does not behave the same as when you leverage kustomize directly.

**Comment by jeffmccune** [comment]
Thanks for reporting this.  I'll look into it.

Some initial thoughts:

I wonder if I used an older version of the upstream type definitions.  This does raise the question, what is the "correct" version?  Is there such a thing?

Ideally upstream would have provided versioned api types, but they didn't given the import path.  That's not unusual, so we do need to figure out how to handle this well for multiple users over time.

**Comment by jeffmccune** [comment]
@brenix I think the right trade-off is to fix this specific issue with a default value which will make the JSON encoder happy.  See below for my proposed fix in the tool and a technique you can use to work-around issues like this without waiting on a release.

I've run into this class of issue before.  This default value technique can be re-used for any imported schema that expects a concrete value but CUE imports as a `string`, which then becomes problematic for Marshal().

We can fix this by unifying that particular field with `string | *""`.  CUE supports both you and I making this change without having to manually edit the generated files.  For me, CUE unifies `cue.mod/pkg/sigs.k8s.io/kustomize/api/types/*.cue` files.  For you, CUE unifies `cue.mod/usr/sigs.k8s.io/kustomize/api/types/*.cue` files.

Note the difference of `usr` for you, `pkg` for me, and `gen` for what cue generated.

If you add the following file and revert your edit, it should work as you expect:

```
cue.mod/usr/sigs.k8s.io/kustomize/api/types/var.cue
```

```cue
package types

#Target: {
	// Ensure name has a concrete value so json.Marshal works.  See
	// https://github.com/holos-run/holos/issues/348
	name: string | *""
}
```

I'm going to add a test and do the same in the `pkg` path and cut a release if it works.

The (much) bigger question of "how do we version unversioned schemas like kustomize?" can be addressed separately as we learn more about what works and what doesn't.  My initial plan is to import this class of schemas into specific, versioned import paths to support multiple users and use cases over time.

**Comment by jeffmccune** [comment]
Fixed in #351 released in `v0.100.1`, `brew install holos-run/tap/holos` to update.
